### PR TITLE
Upgrate to rubocop 0.85.0 and enable `Bundler/GemComment` cop

### PR DESCRIPTION
**Pull request by ric2b**
- Upgrate to `rubocop` v0.85.0
- Enable enforcement of Gem comments when specifying versions or sources 
- Drop support for Ruby 2.3, as rubocop v0.81+ doesn't support it anymore

prime: @jturkel

### CODE REVIEW

**Comment by ric2b** [comment]
Do we still use Ruby 2.3? Those are the tests that are failing.

Ruby 2.3 has not been supported by the maintainers for over a year: https://www.ruby-lang.org/en/news/2019/03/31/support-of-ruby-2-3-has-ended/

File: conf/rubocop_without_rspec.yml
Comment (jturkel) [inline conf/rubocop_without_rspec.yml:16]: Do we need comments for source? Won't the usually be for a repo with a bundled gem?

**Review by jturkel — COMMENTED** [review]
While we're in here how do you feel about make a few changes to our defaults that most projects seem to have settled on?

```yaml
Layout/LineLength:
  Enabled: true
  Max: 120
  IgnoreCopDirectives: true
  IgnoredPatterns:
  - ^#

Style/FrozenStringLiteralComment:
  Enabled: true
```

File: conf/rubocop_without_rspec.yml
Comment (ric2b) [reply conf/rubocop_without_rspec.yml:16]: Just trying to cover the several ways of changing a gem's source.

**Comment by ric2b** [comment]
> While we're in here how do you feel about make a few changes to our defaults that most projects seem to have settled on?
> 
> ```yaml
> Layout/LineLength:
>   Enabled: true
>   Max: 120
>   IgnoreCopDirectives: true
>   IgnoredPatterns:
>   - ^#
> 
> Style/FrozenStringLiteralComment:
>   Enabled: true
> ```

Done in https://github.com/salsify/salsify_rubocop/pull/32, I had forgotten about that PR 🙈

**Comment by jturkel** [comment]
> Done in #32, I had forgotten about that PR 🙈

Haha. Me too!

File: conf/rubocop_without_rspec.yml
Comment (jturkel) [reply conf/rubocop_without_rspec.yml:16]: Oops. I was confusing this with the `path` option. Seems like we'd only use `source` if there was a gem name conflict between one of our gems in gems.salsify.com and a different gem in rubygems.org and we wanted to pick the rubygems.org version... Definitely a scenario that would warrant a comment!

File: CHANGELOG.md
Comment (jturkel) [inline CHANGELOG.md:6]: Should we add a `spec.required_ruby_version = '>= 2.3'` statement to the gemspec to enforce this?

**Review by jturkel — APPROVED** [review]
One question but looks good. Thanks for making this change!

File: CHANGELOG.md
Comment (ric2b) [reply CHANGELOG.md:6]: Done. What happened to the Travis checks though? 🤔

Comment (jturkel) [reply CHANGELOG.md:6]: Weird. I updated the branch protection rules to require a green Travis build before merging.

**Comment by ric2b** [comment]
@jturkel This is still stuck waiting for travis-ci, should I do a useless commit to see if it goes through?

**Comment by jturkel** [comment]
@ric2b - Yeah. Try pushing an empty commit to see if that get things unstuck.

**Pull request merged**
